internal/auth/providers/google.go

```diff
@@ -119,7 +119,18 @@ func (p *GoogleProvider) SetStatsdClient(statsdClient *statsd.Client) {
 
 // ValidateSessionState attempts to validate the session state's access token.
 func (p *GoogleProvider) ValidateSessionState(s *sessions.SessionState) bool {
-	return validateToken(p, s.AccessToken, nil)
+	if s.AccessToken == "" {
+		return false
+	}
+
+	params := url.Values{"access_token": {s.AccessToken}}
+	endpoint := p.ValidateURL.String() + "?" + params.Encode()
```

Comment:
hmm this is interesting.

As written, this isn't particularly safe if the validate url already has query params added to it. Modifying ValidatURL inline is also not ideal since it's a shared object

I think what we want here is to copy the ValidateURL object and then mutate the query like so: 
```golang
endpoint := &net.URL{}
reflect.DeepCopy(p.ValidateURL, endpoint)
q := endpoint.Query()
q.Add("access_token", s.AccessToken)
endpoint.RawQuery = q.Encode()
```

Reply:
we could simplify this a bit by doing a simple pointer shallow copy also fwiw
```golang
var endpoint url.Url
endpoint = *p.ValidateURL
q := endpoint.Query()
q.Add("access_token", s.AccessToken)
endpoint.RawQuery = q.Encode()
```

Reply:
This is a great catch - thank you for the detailed code snippets!